test/test_closure_pte.cpp

```diff
@@ -0,0 +1,143 @@
+//------------------------------------------------------------------------------
+// © 2021-2023. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+
+#include <array>
+#include <string>
+
+#ifndef CATCH_CONFIG_FAST_COMPILE
+#define CATCH_CONFIG_FAST_COMPILE
+#include <catch2/catch_test_macros.hpp>
+#endif
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-eos/closure/mixed_cell_models.hpp>
+#include <singularity-eos/eos/eos.hpp>
+#include <test/eos_unit_test_helpers.hpp>
+
+using namespace singularity;
+
+constexpr Real GPa = 1.0e10;
+constexpr Real MJ_per_kg = 1.0e10;
+
+#ifdef SINGULARITY_BUILD_CLOSURE
+#ifdef SINGULARITY_TEST_SESAME
+#ifdef SINGULARITY_USE_SPINER
+#ifdef SPINER_USE_HDF
+
+SCENARIO("Density-Temperature PTE Solver", "[PTE]") {
+
+  GIVEN("Equations of state") {
+    // Set up the three EOS
+    constexpr int num_eos = 3;
+    std::array<EOS, num_eos> eos_arr;
+    // Reference state
+    constexpr Real P0 = 1.0e6; // 1 bar
+    constexpr Real T0 = 296;   // K
+    // Ideal gas air
+    constexpr Real rho0_air = 1e-03; // g/cc
+    constexpr Real Gruneisen_air = 0.4;
+    constexpr Real CV_air = P0 / rho0_air / (Gruneisen_air * T0);
+    eos_arr[0] = IdealGas(Gruneisen_air, CV_air);
+    // Spiner copper EOS
+    constexpr int Cu_matid = 3337;
+    const std::string eos_file = "../materials.sp5";
+    eos_arr[1] = SpinerEOSDependsRhoT(eos_file, Cu_matid);
+    // Davis Products EOS
+    constexpr Real a = 0.798311;
+    constexpr Real b = 0.58;
+    constexpr Real k = 1.35;
+    constexpr Real n = 2.66182;
+    constexpr Real vc = 0.75419;              // cm^3 / g
+    constexpr Real pc = 3.2 * GPa;            // microbar
+    constexpr Real Cv = 0.001072 * MJ_per_kg; // erg / g / K
+    constexpr Real Q = 4.115 * MJ_per_kg;
+    eos_arr[2] = DavisProducts(a, b, k, n, vc, pc, Cv).Modify<ShiftedEOS>(-Q);
+
+    // Move EOS array from host to device
+    for (auto eos : eos_arr) {
+      eos.GetOnDevice();
```

Comment:
Recall that get on device is *not* a void function. It returns a new eos that needs to be assigned @jhp-lanl

Reply:
I thought maybe the copy I'm doing later would be equivalent, but I suppose I need to return the new EOS on host and then copy that to device rather than copying the old EOS object to device.

Reply:
I skipped the `portableCopyToDevice` and instead just put the output of `GetOnDevice()` in the device array. Is this what you meant?

Reply:
Okay so keep the output of `GetOnDevice()` in the host array and then copy that array to device? I'll change it

Reply:
Alright this should be fixed now

Reply:
I redid this again after a discussion with @dholladay00 about needing to call `Finalize()` on both the original object and the one that gets created when `GetOnDevice()` is called.